src/main/java/ga/rugal/food/core/service/impl/MenuServiceImpl.java

```diff
@@ -50,5 +53,18 @@ public Menu deleteById(Integer id)
     {
         return menuDao.deleteById(id);
     }
+    
+    @Override
+    public List<Menu> getMenusByRestaurant(Restaurant r) {
+        return menuDao.getMenusByRestaurant(r);
+    }
+    
+    @Override
+    public Menu getRandomMenuByRetaurant(Restaurant r) {
+        Random rd = new Random();
```

Comment:
Rethink about repeating initialization of Random object. What about using a field member. or static object.